src/gt4py/next/iterator/ir_utils/common_pattern_matcher.py

```diff
@@ -32,6 +31,19 @@ def is_let(node: itir.Node) -> TypeGuard[itir.FunCall]:
     return isinstance(node, itir.FunCall) and isinstance(node.fun, itir.Lambda)
 
 
-def is_if_call(node: itir.Expr) -> TypeGuard[itir.FunCall]:
-    """Match expression of the form `if_(cond, true_branch, false_branch)`."""
-    return isinstance(node, itir.FunCall) and node.fun == im.ref("if_")
+def is_call_to(node: itir.Node, fun: str | list[str]) -> TypeGuard[itir.FunCall]:
```

Comment:
Just an **optional** suggestion, feel free to ignore: I'd like better to include the actual name of the node `FunCall` in its typeguard function (e.g. `is_funcall_to`).

Reply:
We opted for maximum convenience in the ir_makers module, e.g. `im.ref` also creates a `SymRef` so I'll keep `call` for now.